com.playeveryware.eos/Runtime/Core/Config/Attributes/FieldValidator.cs

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2024 PlayEveryWare
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+namespace PlayEveryWare.EpicOnlineServices
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Reflection;
+
+    public struct FieldValidatorFailure
+    {
+        public FieldInfo FieldInfo;
+        public FieldValidatorAttribute FailingAttribute;
+        public string FailingMessage;
+
+        public FieldValidatorFailure(FieldInfo failingField,
+            FieldValidatorAttribute failingAttribute,
+            string failingMessage)
+        {
+            FieldInfo = failingField;
+            FailingAttribute = failingAttribute;
+            FailingMessage = failingMessage;
+        }
+    }
+    public class FieldValidator
+    {
+        public static bool TryGetFailingValidatorAttributes(object target, out List<FieldValidatorFailure> failingValidatorAttributes)
```

Comment:
Something possibly lost from this change is that it isn't checking fields inside itself that also have attributes. Should the config validator be able to do that all in one call? Is there a way we can recursively ask children for their attributes?

Reply:
Yes, this is next task on deck.